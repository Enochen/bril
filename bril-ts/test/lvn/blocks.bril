@main {
  a: int = const 4;
  b: int = const 2;
  sum1: int = add a b;
  sum2: int = add b a;
.idk:
  prod1: int = mul sum1 sum2;
.idk2:
  prod1: int = mul a b;
  prod2: int = mul sum2 sum1;
  prod1: int = mul sum1 sum2;
  print prod1;
  prod2: int = mul prod2 sum1;
  print prod2;
}
