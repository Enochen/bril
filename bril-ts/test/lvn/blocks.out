@main {
  sum1: int = const 6;
  sum2: int = const 6;
.idk:
.idk2:
  lvn_temp_1: int = mul sum2 sum1;
  print lvn_temp_1;
  prod2: int = mul lvn_temp_1 sum1;
  print prod2;
}
